test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -155,6 +174,244 @@ public void testLogReplicationFSMTransitions() throws Exception {
+     * for a transition from initialized to snapshot sync state.
+     *
+     */
+    @Test
+    public void testSyncStatusUpdatesForSnapshot() throws Exception {
+        final int updateToStatusTableFromOnEntry = 3;
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
+
+        CountDownLatch statusTableLatch = new CountDownLatch(updateToStatusTableFromOnEntry);
+        TestStatusTableStreamListener streamListener = new TestStatusTableStreamListener(statusTableLatch);
+        corfuStore.subscribeListener(streamListener, NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        // Initial state: Initialized
+        LogReplicationState initState = fsm.getState();
+        assertThat(initState.getType()).isEqualTo(LogReplicationStateType.INITIALIZED);
+
+        transitionAvailable.acquire();
+
+        // Transition #1: SNAPSHOT Sync Start
+        transition(LogReplicationEventType.SNAPSHOT_SYNC_REQUEST, LogReplicationStateType.IN_SNAPSHOT_SYNC);
+
+        // Await for update to status table from entry of InSnapshotSyncState, an update stemming
+        // from the SYNC_CANCEL request, and an additional update from the TsPollingTask
+        statusTableLatch.await();
+        corfuStore.unsubscribeListener(streamListener);
+
+        // Sync type should be snapshot for all three updates
+        Set<SyncType> expectedSyncTypes = new HashSet<>(Collections.singletonList(SyncType.SNAPSHOT));
+        // Sync status for both outer and inner should be ongoing for all three updates
+        Set<SyncStatus> expectedSyncStatus = new HashSet<>(Collections.singletonList(SyncStatus.ONGOING));
+        Set<SyncType> actualSyncTypes = new HashSet<>();
+        Set<SyncStatus> actualSyncStatus = new HashSet<>();
+        Set<SyncStatus> actualSnapshotInfoSyncStatus = new HashSet<>();
+
+        Iterator<CorfuStreamEntries> entriesIterator = streamListener.getEntries().iterator();
+        while (entriesIterator.hasNext()) {
+            for (List<CorfuStreamEntry> entry : entriesIterator.next().getEntries().values()) {
+                ReplicationStatusVal replicationStatusVal = (ReplicationStatusVal) entry.get(0).getPayload();
+                actualSyncTypes.add(replicationStatusVal.getSyncType());
+                actualSyncStatus.add(replicationStatusVal.getStatus());
+                actualSnapshotInfoSyncStatus.add(replicationStatusVal.getSnapshotSyncInfo().getStatus());
+            }
+        }
+
+        Assert.assertEquals(expectedSyncTypes, actualSyncTypes);
+        Assert.assertEquals(expectedSyncStatus, actualSyncStatus);
+        Assert.assertEquals(expectedSyncStatus, actualSnapshotInfoSyncStatus);
+    }
+
+    /**
+     * Verify the updates to the status table accurately reflect the sync type and sync status
+     * for a transition from initialized to log entry state.
+     *
+     */
+    @Test
+    public void testSyncStatusUpdatesForLogEntry() throws Exception {
+        final int updateToStatusTableFromOnEntry = 2;
```

Comment:
wouldnt a SYNC_CANCEL be received for log entry sync as well?

Reply:
SYNC_CANCEL will be triggered and it will cause an entry in snapshot sync state.  The behavior on entering this state has been verified by tests in this PR.